daemon/log.c

```diff
@@ -447,60 +449,35 @@ log_logmask(void)
 //
 //
 
-#define BACKTRACE_SIZE 10
-#define FILL_RETURN_ADDRESS(x)                          \
-    if (__builtin_frame_address(x) == 0)                \
-    {                                                   \
-        return_address[x] = NULL;                       \
-        goto end_fill;                                  \
-    }                                                   \
-    return_address[x] = __builtin_return_address(x);    \
-
-        
+#define BACKTRACE_SIZE 10        
 
 void
 log_backtrace(MTC_S32 priority)
 {
     int level;
-    Dl_info dli;
-    void *return_address[BACKTRACE_SIZE] = {NULL};
-
-    level = 0;
-
-    FILL_RETURN_ADDRESS(0);
-    FILL_RETURN_ADDRESS(1);
-    FILL_RETURN_ADDRESS(2);
-    FILL_RETURN_ADDRESS(3);
-    FILL_RETURN_ADDRESS(4);
-    FILL_RETURN_ADDRESS(5);
-    FILL_RETURN_ADDRESS(6);
-    FILL_RETURN_ADDRESS(7);
-    FILL_RETURN_ADDRESS(8);
-    FILL_RETURN_ADDRESS(9);
- end_fill:
-
+    void* return_address[BACKTRACE_SIZE] = {NULL};
+    
     log_message(priority, "backtrace -------------\n");
+    int nptrs = backtrace(return_address, sizeof(return_address) / sizeof(return_address[0]));
```

Comment:
Using `BACKTRACE_SIZE` seems preferable to repeating `sizeof(return_address) / sizeof(return_address[0])` a few times.